src/main/java/seedu/duke/operationlist/FlightList.java

```diff
@@ -0,0 +1,344 @@
+        getFlightNumber(flightDetail);
+        getAirline(flightDetail);
+        getDestination(flightDetail);
+        getDepartureTime(flightDetail);
+        getGateNumber(flightDetail);
+        getCheckIn(flightDetail);
+    }
+
+    private void getFlightNumber(String detail) throws SkyControlException {
+        if (isAdd) {
+            flightNumber = extractDetail(detail, FLIGHT_NUMBER_DELIMITER, AIRLINE_DELIMITER).toUpperCase();
+        }
+    }
+
+    private void getAirline(String detail) throws SkyControlException {
+        airline = extractDetail(detail, AIRLINE_DELIMITER, DESTINATION_DELIMITER).toUpperCase();
+    }
+
+    private void getDestination(String detail) throws SkyControlException {
+        destination = extractDetail(detail, DESTINATION_DELIMITER, DEPARTURE_TIME_DELIMITER).toUpperCase();
+    }
+
+    private void getDepartureTime(String detail) throws SkyControlException {
+        departureTime = extractDetail(detail, DEPARTURE_TIME_DELIMITER, GATE_NUMBER_DELIMITER).toUpperCase();
+    }
+
+    private void getGateNumber(String detail) throws SkyControlException {
+        gateNumber = extractDetail(detail, GATE_NUMBER_DELIMITER, CHECK_IN_ROW_DELIMITER).toUpperCase();
+    }
+
+    private void getCheckIn(String detail) throws SkyControlException {
+        int indexOfCheckIn = detail.indexOf(CHECK_IN_ROW_DELIMITER);
+        int startIndex = indexOfCheckIn + CHECK_IN_ROW_DELIMITER.length();
+        checkIn = detail.substring(startIndex).toUpperCase();
+        if (checkIn.equals(EMPTY_STRING)) {
+            throw new SkyControlException(ui.getMissingDetailsError());
+        }
+    }
+
+    private void validateDetailFormat() throws SkyControlException {
+        validateAirlineLength(airline);
+        validateDestinationLength(destination);
+        validateTime(departureTime);
+        validateCheckIn(checkIn);
+        validateGateNumber(gateNumber);
+    }
+
+    private void checkFlightNumberDuplicates() throws SkyControlException {
+        for (int i = 0; i < flightIndex; i++) {
+            validateFlight(i);
+            if (isFlightDuplicate()) {
+                resetChecks();
+                throw new SkyControlException(ui.getDuplicateFlightError());
+            }
+        }
+    }
+
+    private void checkAvailableGateNumber() throws SkyControlException {
+        for (int i = 0; i < flightIndex; i++) {
+            validateFlight(i);
+            if (isGateOccupied()) {
+                resetChecks();
+                throw new SkyControlException(ui.getGateOccupiedError());
+            }
+        }
+    }
+
+    private void validateFlight(int index) {
+        getNumberOfFlights();
+        assert index < numOfFlights;
+        isFlightNumberPresent = flights.get(index).getFlightNumber().equals(flightNumber);
+        isDepartureTimePresent = flights.get(index).getDepartureTime().contains(departureTime);
+        isGateNumberPresent = flights.get(index).getGateNum().contains(gateNumber);
```

Comment:
Different levels of abstraction in a method. Consider SLAP more and put the logic into another method.